src/survey.ts

```diff
@@ -0,0 +1,212 @@
+// ----------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+// ----------------------------------------------------------------------------
+
+import { commands, MessageItem, Uri, window, workspace } from 'vscode';
+import { callWithTelemetryAndErrorHandling, IActionContext } from "vscode-azureextensionui";
+import { configPrefix } from './constants';
+import { ext } from "./extensionVariables";
+import { assert } from './fixed_assert';
+import { httpGet } from "./httpGet";
+import { hoursToMs, minutesToMs, weeksToMs } from "./util/time";
+
+namespace stateKeys {
+    export const neverShowSurvey = 'neverShowSurvey';
+    export const surveyPostponedUntilTime = 'surveyPostponedUntilTime';
+}
+
+const linkToSurvey = 'https://aka.ms/arm-tools-survey';
+const surveyPrompt = "Could you please take 2 minutes to tell us how well the Azure Resource Manager Tools extension is working for you?";
+
+// Wait for this amount of time since extension first used (this session) before
+// considering asking the survey (only if still interacting with extension)
+interface ISurveyConstants {
+    activeUsageTimeBeforeSurvey: number;
```

Comment:
>activeUsageTimeBeforeSurvey [](start = 4, length = 27)

Nit: maybe add Ms suffix to these, though it's kind of obvious from the usage immediately below.